File: app/src/main/java/com/wang/mytest/MainFragment.java
package com.wang.mytest;

import android.animation.AnimatorInflater;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;

import com.wang.mytest.common.acitivity.ActivityFilter;
import com.wang.mytest.common.util.ActivityUtils;

import java.util.List;
import java.util.concurrent.atomic.AtomicBoolean;

public class MainFragment extends Fragment {
    private static final String TAG = "MainFragment";

    public static MainFragment newInstance() {
        return new MainFragment();
    }

    private final AtomicBoolean mClickable = new AtomicBoolean(true);

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        FrameLayout rootView = new FrameLayout(requireContext()) {
            @Override
            public boolean onInterceptTouchEvent(MotionEvent ev) {
                if (!mClickable.get()) {
                    return true;
                } else {
                    return super.onInterceptTouchEvent(ev);
                }
            }
        };
        inflater.inflate(R.layout.fragment_main, rootView, true);
        return rootView;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        final Context context = requireContext();
        Toolbar toolbar = view.findViewById(R.id.main_toolbar);
        ViewGroup container = view.findViewById(R.id.action_container);

        toolbar.setTitle("MyTest");
        toolbar.inflateMenu(R.menu.main);
        toolbar.setOnMenuItemClickListener(this::onMenuItemSelected);

        List<ActivityFilter.Entity> activities = ActivityFilter.getActivities();
        for (ActivityFilter.Entity entity : activities) {
            final TextView child = createChild(context);
            child.setText(entity.getSummary());
            child.setOnClickListener(v -> {
                ComponentName componentName = new ComponentName(entity.getPackageName(), entity.getClassName());
                Intent intent = new Intent().setComponent(componentName);
                ActivityUtils.startActivity(context, intent);
            });
            container.addView(child);
        }
    }

    @NonNull
    private TextView createChild(@NonNull Context context) {
        TextView view = new TextView(context);
        view.setTextSize(24);
        view.setGravity(Gravity.CENTER);
        view.setBackgroundColor(Color.parseColor("#88888888"));
        view.setPadding(24, 12, 24, 12);
        view.setStateListAnimator(AnimatorInflater.loadStateListAnimator(context, R.animator.state_list_animator));

        ViewGroup.MarginLayoutParams params = new ViewGroup.MarginLayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
        params.leftMargin = 24;
        params.rightMargin = 12;
        params.topMargin = 24;
        params.bottomMargin = 12;
        view.setLayoutParams(params);
        return view;
    }

    private boolean onMenuItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case R.id.options_clickable: {
                mClickable.compareAndSet(false, true);
                return true;
            }
            case R.id.options_unclickable: {
                mClickable.compareAndSet(true, false);
                return true;
            }
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: app/src/main/java/com/wang/mytest/MainFragment.kt
package com.wang.mytest

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.*
import android.widget.Button
import android.widget.FrameLayout
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.wang.mytest.apt.annotation.RouteBean
import com.wang.mytest.apt.api.RouteStore
import com.wang.mytest.feature.ui.ScreenActivity
import com.wang.mytest.feature.ui.view_pager2.ViewPager2Activity
import com.wang.mytest.lifecycle.LifecycleActivity
import java.util.concurrent.atomic.AtomicBoolean

class MainFragment : Fragment() {

    companion object {
        const val TAG = "MainFragment"

        fun newInstance(): MainFragment {
            return MainFragment()
        }
    }

    private lateinit var mRecyclerView: RecyclerView
    private lateinit var mBtnTest: Button

    private lateinit var mContext: Context
    private val mClickable = AtomicBoolean(true)

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mContext = context.applicationContext
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        Log.d(TAG, "onCreateView: ")
        val rootView = object : FrameLayout(activity!!) {
            override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
                return if (!mClickable.get()) {
                    true
                } else super.onInterceptTouchEvent(ev)
            }
        }
        layoutInflater.inflate(R.layout.fragment_main, rootView, true)
        return rootView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        Log.d(TAG, "onViewCreated: ")
        mBtnTest = view.findViewById(R.id.btn_test)
        mRecyclerView = view.findViewById(R.id.recycler_view)
        mRecyclerView.layoutManager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
        mRecyclerView.itemAnimator = DefaultItemAnimator()

        Log.i(TAG, "onViewCreated: " + RouteStore.getAll())
        mRecyclerView.adapter = NormalAdapter(RouteStore.getAll())

        mBtnTest.setOnClickListener {
            startActivity(Intent().setClass(context!!, LifecycleActivity::class.java))
        }
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        inflater.inflate(R.menu.clickable, menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.options_clickable -> {
                mClickable.compareAndSet(false, true)
                true
            }
            R.id.options_unclickable -> {
                mClickable.compareAndSet(true, false)
                true
            }
            R.id.options_screen_details -> {
                startActivity(Intent(context, ScreenActivity::class.java))
                true
            }
            R.id.options_viewpager2 -> {
                startActivity(Intent(context, ViewPager2Activity::class.java))
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    inner class NormalAdapter(private val mDataList: List<RouteBean>) : RecyclerView.Adapter<NormalAdapter.ViewHolder>() {

        inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
            public val tvTitle: TextView = view.findViewById(R.id.tv_title)
        }

        override fun getItemCount(): Int {
            return mDataList.size
        }

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
            val view = LayoutInflater.from(parent.context).inflate(R.layout.item_1, parent, false)
            return ViewHolder(view)
        }

        override fun onBindViewHolder(holder: ViewHolder, position: Int) {
            holder.tvTitle.text = mDataList[position].title
            holder.itemView.setOnClickListener {
                val routeBean = mDataList[position]
                val intent = Intent().setClassName(this@MainFragment.context!!, routeBean.className)
                startActivity(intent)
            }
        }
    }
}